src/lib/opal_parser.h

```diff
@@ -66,10 +66,17 @@ enum opal_user {
 	OPAL_USER9 = 0x09,
 };
 
+/* The following enum are mapped to
+   Opal read and write locking state,
+   OPAL_RO requires both RL & RLE on a
+   locking range to be enabled in order
+   to be effective. OPAL_WR requires
+   both WL & WLE on a locking range to
+   be enabled in order to be effective */
+
```

Comment:
The description here is really helpful.
Why are we removing the OPAL_LK here? This is required to lock the device from either reading or writing.
By RW, we meant to let the users Read/Write  from/to the device. 
I don't think it makes sense to let the users only write to the device while the read lock is enabled and can't read from the device.

Reply:
@RevanthRajashekar, please have a look of my other general comment in the PR. I can see the current enum definition for sedopal_lock_unlock (which i think is used when sedcli is using with opal driver) make sense. However with non-opal driver (i.e NVMe PT), we sending the tokenized payload to SSD dirctly, there is no direct mapping for OPAL_RW and OPAL_LK to a locking range state defined by Locking SP table in the spec. In current implementation, you are only changing RL/WL, how do you ensure RLE and WLE are set correctly?

Reply:
Maybe enum opal_lock_state is *not* intended to map to Locking SP RL/RLE/WL/WLE after all.

Reply:
The --lock-unlock  functionality is used to Lock or unlock global locking range.
Before using it, make sure that you have setup the global locking range using the --setup-global-range command, which enables those RLE and WLE columns.